### ISSUE TITLE
Color Stack Behaviour

### ISSUE THREAD

**Issue by patrick96** [jaagr/polybar#639]
In f754fa27103e0df9a10b0083babd7ad29605e016 you introduced color stacks to fix the `-prefix` color problems described in #544.
With those color stacks the following module will make subsequent modules use `#373B41` as the background color unless they explicitly specify another color.
```dosini
[module/text]
type = custom/text
content = "%{B#373B41}1%{B#DE935F}3%{B-}"
```
Is this now intended behaviour and are format tags now supposed to be used in pairs (like HTML tags) so that after a module is processed the stack is empty again?

If this isn't intended behaviour, then maybe stacks are not the solution to this problem. It would probably be better to inject the color tags for the main section in between the prefix and the main section of the module as well as before the whole module. Or determine the colors for the prefix and main sections seperately and putting the prefix tags only in front of the prefix section and the same for the main section with the prefix section having two fallbacks: the color of the main section and the default color for the bar.

If this is intended behaviour, then there should probably be a way to totally reset the format tags (not just a simple reset that merely pops a value from the stack).

With the latter I may be able to help and submit a PR but for the former I have no idea how to do this, since I haven't figured out where and how the prefixes are handled

EDIT: See #666, #684, #575,  #736  for how this affects progressbars (the progressbar doesn't properly close its color tags)

**Comment by NBonaparte** [comment]
From what I read in #544, this is intended (after all, it is a "stack").

> We'll have to create a value stack for some formatting tags to be able to pop to the previous value because now the resetting (e.g. %{F-}) only sets the default value defined for the bar.

I think we should just implement a full reset tag which pops all of the values except the bottom.
The problem is that the current system diverges from lemonbar functionality, so I'm not sure if we should revert the `-` symbol to being the "restore to original" symbol and use a different one for popping only one item off the stack, or abandon the lemonbar syntax and use a different symbol for the full reset.

Prefixes should be handled in [src/modules/meta/base.cpp](https://github.com/jaagr/polybar/blob/master/src/modules/meta/base.cpp) (see ffbedf4).

**Comment by patrick96** [comment]
Yeah I read that too, but I wondered if maybe the problems that occur now with the stack are unintended sideeffects and the fix for #544 is just some rewriting of how prefixes and suffixes are handled.

If the stack and all its sideeffects were actually intended then I think we should not break existing configs and scripts and should keep `%{F-}` as a total reset (to bar FG) and use some other syntax for the closing tag (maybe something like `%{F!}`).

Personally I think we should revert the stack changes and directly fix the prefix color issue, since I don't see any other benefit from using a color stack and it just brings more problems with it.
